src/main/groovy/io/seqera/wave/service/packages/AbstractPackagesService.groovy

```diff
@@ -83,17 +83,23 @@ abstract class AbstractPackagesService implements PackagesService{
         try(final reader = Files.newBufferedReader(file)) {
             String line
             boolean begin=false
-            List<CondaPackageRecord> entries = new ArrayList<>()
+            //Map to keep track of packages
+            Map<String, CondaPackageRecord> packages = new HashMap<>()
```

Comment:
This means we need to load \*all* packages in memory, and this list can be very big. To handle it properly it would be required a different approach, and I think it makes no sense to spend time on this now. I'd say to revert this

Reply:
Currently, we are using ArrayList to load all packages.

Reply:
done